### PR TITLE
Handle provided fields and nested keys better

### PR DESCRIPTION
**Pull request by ardatan**
Refactor handling `@provides` directive

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,9 @@
     "test": "vitest --project unit",
     "test:bun": "bun test --bail",
     "test:e2e": "vitest --project e2e",
-    "test:leaks": "cross-env \"LEAK_TEST=1\" jest --no-watchman --runInBand --detectOpenHandles --detectLeaks --logHeapUsage"
+    "test:leaks": "cross-env \"LEAK_TEST=1\" jest --no-watchman --runInBand --detectOpenHandles --detectLeaks --logHeapUsage",
+    "build:alpha": "yarn build && yarn pin-workspace-deps",
```
Comment (enisdenjo) [inline package.json]: Each package has a `prepack` script which will build only the package being released. So building all packages is completely unnecessary.

```suggestion
    "build:alpha": "yarn pin-workspace-deps",
```

File: package.json [hunk 2]
```diff
@@ -22,7 +22,9 @@
...
+    "pin-workspace-deps": "node ./scripts/pin-workspace-deps.mjs"
```
Comment (enisdenjo) [inline package.json]: You should write the scripts in typescript, add them to the `include` in tsconfig and execute with tsx. This way everything is kept typesafe.

**Review by enisdenjo — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -6,7 +6,9 @@ on:
       - main
   pull_request:
 
-concurrency: ${{ github.workflow }}-${{ github.ref }}
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (enisdenjo) [inline .github/workflows/release.yml:11]: Not sure how safe it is to cancel a running release workflow. 🤔 

But I guess, even if the release did not complete altogether - the release CI that superseeds should.

Comment (ardatan) [reply .github/workflows/release.yml]: Reverted 👍

**Review by enisdenjo — APPROVED** [review]

**Pull request merged**